x5.md

```diff
@@ -8,7 +8,118 @@ created: 2019-04-18
+- In order to listen to a public group channel, a client must subscribe to the channel name.  The sender of a message is derived from the message's signature.
+- Discovery of channel names is not currently part of the protocol, and is typically done out of band.  If a channel name is used that has not been used, it will be created.
+- channel name specification:
+    - lower case alphanumeric
+    - no special characters excluding `-`
+    - TODO: other langauge characters
+
+#### 1.2 Private 1:1 messages
+This can be done in a the following ways:
+1. scanning a user generated QR code
+1. discovery through the Status app
+1. asyncronous X3DH key exchange
+1. public key via public channel listening
+    - `status-react/src/status_im/contact_code/core.cljs`
+1. contact codes
+2. decentralized storage (not implemented)
+3. whisper
+
+### 2. Initial Key Exchange
+
+#### Contact Request
+
+#### Bundles
+- NOTE: cant seem to find bundle code on account generation
+- An X3DH prekey bundle is defined as ([code(https://github.com/status-im/status-go/messaging/chat/protobuf/encryption.pb.go)]):
+  ```
+  Identity                // Identity key
+  SignedPreKeys           // a map of installation id to array of signed prekeys by that installation id
+  Signature               // Prekey signature
+  Timestamp               // When the bundle was lasted created locally
+  XXX_NoUnkeyedLiteral    // ???
+  XXX_unrecognized        // ???
+  XXX_sizecache           // ???
+  ```
+  - include BundleContainer???
+- a new bundle SHOULD be created at least every 12 hours
+#### QR code
+- A generated QR code should include a X3DH bundle set along with the contact code but I can't find the code to do so.
+
+### 4. Contact Verification
+Once you have the information of a contact, the following can be used to verify that the key material is as it should be.
+#### Identicon
+A low-poly identicon is deterministically generated from the whisper chat public key.  This can then be compared out of band to ensure the reciever's public key is the one you have locally.
+#### 3 word pseudonym / whisper key fingerprint
+Status generates a deterministic 3-word random pseudonym from the whisper chat public key.  This pseudonym acts as a human readable fingerprint to the whisper chat public key.  This name also shows when viewing a contact's public profile and in the chat UI.
+- implementation: [gfycat](https://github.com/status-im/status-react/tree/develop/src/status_im/utils/gfycat)
+#### ENS name
+Status offers the ability to register a mapping of a human readable subdomain of `stateofus.eth` to their whisper chat public key.  This registration is purchased (currently by staking 10 SNT) and stored on the Ethereum mainnet blockchain for public lookup.
+
+## Security Implications
+ 
+ 
+# Previous stuff (introduction)
```

Comment:
Should we just cut all of this?

Reply:
Below I mean, see preview/orig file

Reply:
yeah, we can add it back in appropriately later